justfile

```diff
@@ -152,12 +152,58 @@ web-dev-up:
     rm "./web-api/assets" || :
     ln -s "./assets/smbert_v0003" "./web-api/assets"
     compose="$(command -v podman-compose || command -v docker-compose)"
-    $compose -f "./web-api/compose.yml" up --detach --remove-orphans
+    $compose -f "./web-api/compose.db.yml" up --detach --remove-orphans
 
 web-dev-down:
     #!/usr/bin/env -S bash -eux -o pipefail
     compose="$(command -v podman-compose || command -v docker-compose)"
-    $compose -f "./web-api/compose.yml" down
+    $compose -f "./web-api/compose.db.yml" down
+
+build-service-image $CRATE_PATH $BIN $ASSET_DIR="":
+    #!/usr/bin/env -S bash -eux -o pipefail
+    ociBuilder="$(command -v podman || command -v docker)"
+    # if this is further nested add more .. to the soft-link below
+    out=.tmp/oci-build-dir
+    [ -d "$out" ] && rm -r "$out"
+    mkdir -p "$out"
+    cargo install \
+        --path "$CRATE_PATH" \
+        --bin "$BIN" \
+        --debug \
+        --root "$out"
+    # rename binary as to not needing to modify the Dockerfile
+    mv "$out/bin/$BIN" "$out/server.bin"
+    rmdir "$out/bin"
+    if [ -z "$ASSET_DIR" ]; then
+        ASSET_DIR="$out/assets"
+        mkdir -p "$out/assets"
```

Comment:
Can't we create a hard link instead of the soft link we did previously? The data of the project will not be split across partitions and it works on all platforms. This way we can cut arguments to the Dockerfile that can just copy the whole context.

Reply:
Actually, under docker, setting the context to `.` means that the whole project is zipped and sent to the daemon, so copying the asset in `$out` would be cheaper if hard linking is not possible.

Reply:
> Can't we create a hard link instead 

Hard links don't work for directories. And if they would using `rm -r` would delete the original asset files.

> Actually, under docker, setting the context to . means that the whole project is zipped and sent to the daemon, 

reminds me of why I'm not using docker (but podman instead). I changed it accordingly.